agent/src/beerocks/slave/backhaul_manager/backhaul_manager_thread.cpp

```diff
@@ -1722,6 +1722,60 @@ bool backhaul_manager::handle_slave_backhaul_message(std::shared_ptr<SSlaveSocke
         }
         break;
     }
+    case beerocks_message::ACTION_BACKHAUL_CLIENT_ASSOCIATED_NOTIFICATION: {
+        LOG(DEBUG) << "ACTION_BACKHAUL_CLIENT_ASSOCIATED_NOTIFICATION received from iface "
+                   << soc->hostap_iface;
+        auto msg =
+            beerocks_header
+                ->addClass<beerocks_message::cACTION_BACKHAUL_CLIENT_ASSOCIATED_NOTIFICATION>();
+        if (!msg) {
+            LOG(ERROR) << "Failed building ACTION_BACKHAUL_CLIENT_ASSOCIATED_NOTIFICATION message!";
+            return false;
+        }
+
+        // Create a new tuple to contain client association information
+        auto associated_client =
+            std::make_tuple(msg->client_mac(), std::chrono::steady_clock::now());
+
+        // If exists, remove current client association information for given client
+        auto &&associated_clients =
+            m_radio_info_map[msg->iface_mac()].associated_clients_map[msg->bssid()];
+        auto it =
+            std::find(associated_clients.begin(), associated_clients.end(), associated_client);
+        if (it != associated_clients.end()) {
+            associated_clients.erase(it);
```

Comment:
There is a list of associated clients per BSSID. BSSID value is being received as part of the message and it is used to obtain the list of associated clients to that BSSID. So AFAIK there shouldn't be any problem.

Reply:
I think you missed my point.

When you receive a message that a client has associated with a BSS, you remove that client from any previous BSS it was associated with. My point is that:
- maybe it actually still is associated with that BSS;
- in any case, when it does disassociate, we'll get a disassociation message and it will be removed then.

In other words, it is not needed and maybe incorrect.